src/kademlia/get_item.cpp

```diff
@@ -51,6 +51,9 @@ void get_item::got_data(bdecode_node const& v,
 {
 	// we received data!
 
+	// for put_immutable_item, there is no data_callback.
+	if (!m_data_callback) return;
```

Comment:
The old logic is, we always call m_data_callback for immutable item, for get, it will sent dht_immutable_item_alert, and abort(), for put, it will get the data which going to put to DHT.

This patch add a new interface for put 
```void put_item(sha1_hash const& target, entry& data, boost::function<void(int)> f);```
so we needn't a callback to get data, as result, the m_data_callback is not set, and we can just return here because we don't care about what we get, we only need all nodes to put data.